pkg/controller/core/admissioncheck_controller.go

```diff
@@ -164,7 +164,20 @@ func (r *AdmissionCheckReconciler) Generic(e event.TypedGenericEvent[*kueue.Admi
 func (r *AdmissionCheckReconciler) NotifyClusterQueueUpdate(oldCq *kueue.ClusterQueue, newCq *kueue.ClusterQueue) {
 	log := r.log.WithValues("oldClusterQueue", klog.KObj(oldCq), "newClusterQueue", klog.KObj(newCq))
 	log.V(5).Info("Cluster queue notification")
-	noChange := newCq != nil && oldCq != nil && slices.CmpNoOrder(oldCq.Spec.AdmissionChecks, newCq.Spec.AdmissionChecks)
+
+	// Helper to extract admission check names from strategy
+	getAcNames := func(cq *kueue.ClusterQueue) []kueue.AdmissionCheckReference {
+		if cq.Spec.AdmissionChecksStrategy == nil {
+			return nil
+		}
+		names := make([]kueue.AdmissionCheckReference, len(cq.Spec.AdmissionChecksStrategy.AdmissionChecks))
+		for i, ac := range cq.Spec.AdmissionChecksStrategy.AdmissionChecks {
+			names[i] = ac.Name
+		}
+		return names
+	}
+
+	noChange := newCq != nil && oldCq != nil && slices.CmpNoOrder(getAcNames(oldCq), getAcNames(newCq))
```

Comment:
This change is great and a bugfix, we need also a separete PR we could cherrypick as a bugifx. Feel free to open. In any case keep the changes here as they are.

Reply:
together with the changes in https://github.com/kubernetes-sigs/kueue/pull/7490/files#r2485619146